BlinkID/example/lib/document_scanner.dart

```diff
@@ -0,0 +1,173 @@
+import 'package:blinkid_flutter/microblink_scanner.dart';
+import 'package:blinkid_flutter/microblink_scanner_widget.dart';
+import 'package:flutter/material.dart';
+import 'package:permission_handler/permission_handler.dart';
+
+typedef MicroblinkScannerResultCallback = Future<bool> Function(List<RecognizerResult>?);
+
+typedef MicroblinkRecognizerResult = List<RecognizerResult>;
+
+extension ScanDocumentExtension on BuildContext {
+  Future<MicroblinkRecognizerResult?> scanDocument() async {
+    final permissions = await [Permission.camera].request();
+
+    if (permissions.values.any((status) => status.isDenied)) {
+      ScaffoldMessenger.of(this)
+          .showSnackBar(const SnackBar(content: Text('Permissions are required to scan documents.')));
+    }
+
+    return showDialog(
+      context: this,
+      builder: (_) => const _DocumentScannerDialog(),
+    );
+  }
```

Comment:
Better to separate permissions handling and the showing of the scan dialog. Also, notice how I check that _all_ the permissions are _granted_. `isDeined` is not an exhaustive check, as the permission can be `restricted`, or `permanentlyDenied` for example.

```dart
extension ScanDocumentExtension on BuildContext {
  Future<MicroblinkRecognizerResult?> scanDocument() => _isPermissionGranted().then(
        (isGranted) => isGranted
            ? showDialog(
                context: this,
                builder: (_) => const _DocumentScannerDialog(),
              )
            : () {
                ScaffoldMessenger.of(this).showSnackBar(
                  const SnackBar(
                    content: Text(
                      'Permission to use Camera is required.',
                    ),
                  ),
                );
              }(),
      );

  Future<bool> _isPermissionGranted() =>
      [Permission.camera].request().then((permissions) => permissions.values.every((p) => p.isGranted));
}
```